src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs

```diff
@@ -158,8 +158,22 @@ public bool TryRemoveContext(IAzureContext context)
                             promptAction)?.FirstOrDefault();
                         if (tenant == null || tenant.Id == null)
                         {
-                            throw new ArgumentNullException(string.Format("Could not find tenant id for provided tenant domain '{0}'. Please ensure that " +
-                                                                          "the provided service principal is found in the provided tenant domain.", tenantId));
+                            string baseMessage = string.Format("Could not find tenant id for provided tenant domain '{0}'", tenantId);
```

Comment:
The strings should probably be in Resources.resx.  Other than that looks good to me.